src/Exceptionless.Core/Jobs/Elastic/CleanupSnapshotJob.cs

```diff
@@ -5,19 +5,23 @@
 using Foundatio.Jobs;
 using Foundatio.Lock;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 
 namespace Exceptionless.Core.Jobs.Elastic {
     [Job(Description = "Removes old Elasticsearch snapshots.", IsContinuous = false)]
     public class CleanupSnapshotJob : Foundatio.Repositories.Elasticsearch.Jobs.CleanupSnapshotJob {
-        public CleanupSnapshotJob(ExceptionlessElasticConfiguration configuration, ILockProvider lockProvider, ILoggerFactory loggerFactory)
+        private readonly IOptionsSnapshot<AppOptions> _options;
+
+        public CleanupSnapshotJob(ExceptionlessElasticConfiguration configuration, ILockProvider lockProvider, IOptionsSnapshot<AppOptions> options, ILoggerFactory loggerFactory)
```

Comment:
Why are you using IOptionsSnapshot? We don't support this changing on the fly while the app is running.

Reply:
We have reload=true so we could and it would be nice to support it.

Reply:
It would be a bit scary for some of these things to change on the fly. Seems like we would need to look at all the usages and see which options are accessed many times while the app is running and determine if its a good or bad idea for them to be changing on the fly.